nestkernel/conn_parameter.h

```diff
@@ -145,50 +148,103 @@ class ScalarIntegerParameter : public ConnParameter
 
 
 /**
- * Array parameter, returning values in order.
+ * Array parameter classes, returning double values in order.
  *
  * - The array of values must not be empty
  *   (so return 0 for number_of_values can signal non-array parameter)
  * - Throws exception if more values requested than available.
- *
+ * - The class contains nthread number of pointers (one for each thread)
+ *   to an iterator, which runs over the parameters initialised in an array.
+ *   Each pointer is moved along the paramtere array by the function
+ *   value_double(), which returns the current parameter value and moves the
+ *   pointer to the subsequent position.
+ * - All parameters are  doubles, thus calling the function value_int()
+ *   throws an error.
  */
-class ArrayParameter : public ConnParameter
+
+class ArrayDoubleParameter : public ConnParameter
 {
 public:
-  ArrayParameter( const std::vector< double >& values )
-    : values_( values )
-    , next_( values_.begin() )
+  ArrayDoubleParameter( const std::vector< double >& values, const size_t nthreads )
+    : values_( &values )
+    , next_( nthreads, values_->begin() )
   {
   }
 
   size_t
   number_of_values() const
   {
-    return values_.size();
+    return values_->size();
   }
 
-  // double value(index sgid, index tgid, librandom::RngPtr&) const
   double
-  value_double( index, index, librandom::RngPtr& ) const
-
+  value_double( thread tid, librandom::RngPtr& ) const
   {
-    // return values_[sgid];
-    if ( next_ != values_.end() )
-      return *next_++;
+    if ( next_[ tid ] != values_->end() )
+      return *next_[ tid ]++;
     else
       throw KernelException( "Parameter values exhausted." );
   }
   long_t
-  value_int( index, index, librandom::RngPtr& ) const
+  value_int( thread, librandom::RngPtr& ) const
   {
     throw KernelException( "ConnParameter calls value function with false return type." );
   }
 
 private:
-  std::vector< double > values_;
-  mutable std::vector< double >::iterator next_;
+  const std::vector< double >* values_;
```

Comment:
After offline discussions we came to the conclusion that we shouldn't be concerned about that. The datum is initialised in the conn_builder once and subsequently accessible for all threads via pointers. The value_()-function, which moves the pointer along the parameter array, is called within an omp parallel-block, thus securing that all threads are finished before executing further code.